src/fetchClient.ts

```diff
@@ -0,0 +1,131 @@
+import { ResponseType } from './responseType';
+
+import { FetchClientConfiguration, RequestOptions } from './index';
+type RequestParameters = Record<string, string | number>;
+
+export class FetchClient {
+  private readonly configuration: FetchClientConfiguration = {
+    baseUrl: '',
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    headers: { 'Content-Type': 'application/json' },
+  };
+
+  constructor(configuration: Partial<FetchClientConfiguration> = {}) {
+    this.configuration = {
+      ...this.configuration,
+      ...configuration,
+    };
+  }
+
+  private getPath({
+    path,
+    options,
+  }: {
+    path: string;
+    params?: RequestParameters;
+    options?: RequestOptions;
+  }) {
+    const baseUrl = options?.baseUrl || this.configuration.baseUrl;
+    if (!baseUrl) {
+      throw new Error('FetchClient is not configured. Please pass a baseUrl.');
+    }
+    return [
+      baseUrl.endsWith('/') ? baseUrl.slice(0, -1) : baseUrl,
+      path.startsWith('/') ? path.slice(1) : path,
+    ].join('/');
+  }
+
+  private getHeaders(options?: RequestOptions): Record<string, string> {
+    return {
+      ...this.configuration.headers,
+      ...options?.headers,
+      ...(options?.accessToken
+        ? { Authorization: `Bearer ${options.accessToken}` }
+        : {}),
+    };
+  }
+
+  private static async returnData<T>(response: Response): ResponseType<T> {
+    const text = await response.text();
+    const data = text.length > 0 ? JSON.parse(text) : {};
+    const { headers, ok, redirected, status, statusText, type, url } = response;
```

Comment:
you cannot spread a response 🤷‍♂️  took a while until I found out

Reply:
I would have done it the other way around as this drops all the prototypes of the response. Did not try it but something around the lines of `Object.assing(res, { body: data })`